hail/src/main/scala/is/hail/expr/types/physical/PType.scala

```diff
@@ -323,4 +323,6 @@ abstract class PType extends Serializable with Requiredness {
   final def typeCheck(a: Any): Boolean = a == null || _typeCheck(a)
 
   def _typeCheck(a: Any): Boolean = virtualType._typeCheck(a)
+
+  def load(src: Code[Long]): PValue = PValue(this, Region.loadIRIntermediate(this)(src))
```

Comment:
would like to see Region.loadIRIntermediate removed and this logic implemented on each ptype (probably this is your intention) but not in this PR.

Reply:
Yes, that's 100% my intention.